include/gaussian.h

```diff
@@ -101,6 +125,75 @@ class Gaussian
     ~Gaussian() {};
 };
 
+class Gaussians
+{
+public:
+    typedef std::vector<std::shared_ptr<Gaussian>> Data;
+
+private:
+    Data _data = {};
+
+    size_t assign(const Data * data, size_t i = 0)
+    {
+        size_t i_begin = i;
+        for(const auto & gauss : *data)
+        {
+            if(gauss == nullptr) throw std::runtime_error("ConvolvedGaussian data["
+                + std::to_string(i - i_begin) + "] can't be null");
+            _data[i++] = gauss;
+        }
+        return i;
+    }
+
+public:
+    Gaussian& operator[](size_t i) { return *(_data[i]); }
+    const Gaussian& operator[](size_t i) const { return *(_data[i]); }
+
+    typename Data::iterator begin() noexcept {return _data.begin();}
+    typename Data::const_iterator cbegin() const noexcept {return _data.begin();}
+
+    typename Data::iterator end() noexcept {return _data.end();}
+    typename Data::const_iterator cend() const noexcept {return _data.cend();}
+
+    Data get_data() const { return _data; }
+
+    inline std::shared_ptr<Gaussian> & at(size_t i) {return _data.at(i);}
+    size_t size() const {return _data.size();}
+
+    std::string str() const {
+        std::string s = "Gaussians([";
+        for(const auto & g : _data) s += g->str() + ",";
+        return s + "])";
+    }
+
+    Gaussians(const Data * data)
+    {
+        if(data != nullptr)
+        {
+            size_t n_data = data->size();
+            if(n_data > 0)
+            {
+                _data.resize(n_data);
+                this->assign(data);
+            }
+        }
+    }
+    Gaussians(std::vector<const Data *> data)
```

Comment:
Vectors of raw pointers like this always scare me.  Are the objects these pointers on the stack, typically?  Could you use `std::optional` instead to let you have null values with safer memory management?